### PR TITLE
C1 khloud c3 s13 add words to project

### PR DESCRIPTION
**Pull request by Khloud-Khalid**
[Issue #179 ] @adamggg , @KareemAliRahman , @zoheiry , @nourhanZ

### CODE REVIEW

File: arability/app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -393,27 +391,30 @@ def import_csv
...
+  # params:
+  #   project_id, word_id, synonym_id
+  # success:
+  #   keyword and synonym are added to project and/or synonym of word updated 
+  # failure:
+  #   object not valid (no project or word id), word already exists in project, 
+  #   keyword or synonym does not exist, word add limit exceeded.
+  def add_word_inside_project
+    @project_id = params[:project_id]
+    if Developer.find_by_gamer_id(current_gamer.id) != nil && not(@project_id.blank?)
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: 86 characters

File: arability/app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -393,27 +391,30 @@ def import_csv
...
       @word_id = Keyword.find_by_name(params[:keyword]).id
-      if Keyword.find_by_id(@word_id) != nil
+      if @word_id != nil && Keyword.find_by_id(@word_id) != nil
         @synonym_id = params[:synonym_id]
         if PreferedSynonym.find_word_in_project(@project_id, @word_id)
-          @edited_word = PreferedSynonym.find_by_keyword_id(@word_id) 
+          @edited_word = PreferedSynonym.where(project_id:@project_id,
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Spaces are free, Khloud. You should add one after project_id:

File: arability/app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -393,27 +391,30 @@ def import_csv
...
+    @project_id = params[:project_id]
+    if Developer.find_by_gamer_id(current_gamer.id) != nil && not(@project_id.blank?)
       @word_id = Keyword.find_by_name(params[:keyword]).id
-      if Keyword.find_by_id(@word_id) != nil
+      if @word_id != nil && Keyword.find_by_id(@word_id) != nil
         @synonym_id = params[:synonym_id]
         if PreferedSynonym.find_word_in_project(@project_id, @word_id)
-          @edited_word = PreferedSynonym.find_by_keyword_id(@word_id) 
+          @edited_word = PreferedSynonym.where(project_id:@project_id, 
+            keyword_id:@word_id).first
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Spaces are free.

File: arability/app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -425,20 +426,35 @@ def add_word
...
+            redirect_to :back, flash: flash
             return
           end
         else
-          @added_word = PreferedSynonym.add_keyword_and_synonym_to_project(@synonym_id, @word_id, @project_id)
-          if @added_word
-            flash[:success] = t(:successfully_added_word_to_project)              
-            redirect_to project_path(@project_id), flash: flash
-            return
+          if MySubscription.get_permissions(current_developer.id, 2)
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: What's this 2?

File: arability/app/controllers/projects_controller.rb [hunk 5]
```diff
@@ -425,20 +426,35 @@ def add_word
...
+            if @added_word
+              project_categories = Project.find(@project_id).categories
+              new_keyword = Keyword.find(@word_id)
+              for category in project_categories do
+                if not new_keyword.categories.include?(category) 
+                  new_keyword.categories.push(category)
+                  new_keyword.save
+                end
+              end
+              MySubscription.find_by_developer_id(current_developer.id).decrement_word_add
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Over 80.

File: arability/app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -425,20 +426,35 @@ def add_word
...
-            flash[:success] = t(:successfully_added_word_to_project)              
-            redirect_to project_path(@project_id), flash: flash
-            return
+          if MySubscription.get_permissions(current_developer.id, 2)
+            @added_word = PreferedSynonym.add_keyword_and_synonym_to_project(
+              @synonym_id, @word_id, @project_id)
+            if @added_word
+              project_categories = Project.find(@project_id).categories
+              new_keyword = Keyword.find(@word_id)
+              for category in project_categories do
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Use .each. Ruby is pretty you know.

File: arability/app/controllers/projects_controller.rb [hunk 7]
```diff
@@ -425,20 +426,35 @@ def add_word
...
+                if not new_keyword.categories.include?(category)
```
Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Use ! not not. :laughing:

Comment (adamggg) [inline arability/app/controllers/projects_controller.rb]: Pun intended. :smile:

File: arability/app/controllers/projects_controller.rb (hunk 4 above)
Comment (Khloud-Khalid) [inline arability/app/controllers/projects_controller.rb]: it's the value for add_limit in the MySubscription model

File: arability/app/views/search/search.html.erb [hunk 8]
```diff
@@ -147,6 +148,16 @@
 		<div class="accordion-heading">
 			<span class="accordion-toggle" data-toggle="false" href="#collective">
 				<span><%= t(:view_collectives) %><i><b><%= @search_keyword %></b></i></span>
+				<%= form_tag(add_word_inside_project_path, method: "post", style: "margin:10px; display: inline;") do %>
+					<!-- <input type="hidden" value="<%= @user_id %>" name="user_id"> -->
```
Comment (adamggg) [inline arability/app/views/search/search.html.erb]: Why comment the lines if you don't need them? =D

Comment (Khloud-Khalid) [inline arability/app/views/search/search.html.erb]: Haha :D forgot to erase them

**Comment by krmali** [comment]
@Khloud-Khalid , solve conflicts

**Comment by Khloud-Khalid** [comment]
@KareemAliRahman done.

**Comment by krmali** [comment]
@Khloud-Khalid, you need to specify for the user , whether the word is added of the synonym is changed when displaying the alert, because he/she might have the keyword in his project.

File: arability/app/views/search/search_with_filters.html.erb [hunk 9]
```diff
@@ -45,6 +45,75 @@
 						$(".btnfollow").html("<%= t(:follow) %>");
 					}	
 				}
+				function add_keyword_with_synonym(){
+    				var synonym_id = $("#synonym_select option:selected").val();
+    				var project_id = $("#project_select option:selected").val();
+    				var msg = "batee5";
```
Comment (adamggg) [inline arability/app/views/search/search_with_filters.html.erb]: :watermelon: ?

File: arability/app/views/search/search_with_filters.html.erb [hunk 10]
```diff
@@ -45,6 +45,75 @@
...
+    				if(synonym_id != undefined && synonym_id !="" && project_id != undefined && project_id !=""){
+        				var request = $.post("/developers/projects/"+project_id+"/add_word_inside_project.json", {synonym_id: synonym_id, keyword: "<%= @search_keyword %>", project_id: project_id } );
```
Comment (adamggg) [inline arability/app/views/search/search_with_filters.html.erb]: Are spaces free? Question of wisdom!

File: arability/app/views/search/search_with_filters.html.erb [hunk 11]
```diff
@@ -45,6 +45,75 @@
...
+    							add_showSuccess(project_id, data.toString());
+    						else
+    							add_error(data.toString());
+  						});
+  						request.fail(function (error, _, _){
+    						add_error(error.toString());
+  						});
+      				}
+      				else{
+      					if(project_id == undefined || project_id =="")
```
Comment (adamggg) [inline arability/app/views/search/search_with_filters.html.erb]: Add space after == :space_invader:

File: arability/app/views/search/search_with_filters.html.erb [hunk 12]
```diff
@@ -45,6 +45,75 @@
...
+ 					$("#add_success").css("display", "block");
+  					$("#add_success").css("margin-bottom", "2px");
+  					$("#add_success").css("position", "fixed");
+  					$("#add_success").css("top", "50px");
+  					$("#add_success").css("left", "10%");
+  					$("#add_success").css("right", "10%");
+				}
+				function add_error(msg){
+					var text = document.getElementById("fail_msg");
+  					text.innerHTML = msg;
```
Comment (adamggg) [inline arability/app/views/search/search_with_filters.html.erb]: :shit: Fix indentations =D

**Comment by Khloud-Khalid** [comment]
@adamggg please merge since you're the last reviewer to verify =)

**Pull request merged**